server/aws-lsp-codewhisperer/src/language-server/workspaceContext/dependency/dependencyDiscoverer.ts

```diff
@@ -144,12 +149,45 @@ export class DependencyDiscoverer {
         }
     }
 
-    async handleDependencyUpdateFromLSP(language: string, paths: string[], workspaceRoot?: WorkspaceFolder) {
+    public isDependencyEventsIngested(workspaceFolderUri: string): boolean {
+        return this.dependencyEventsIngestedFolderUris.has(workspaceFolderUri)
+    }
+
+    private ingestRecordedDependencyEvents(workspaceFolders: WorkspaceFolder[]): void {
+        let ingestedDependencyCount = 0
+        for (const workspaceFolder of workspaceFolders) {
+            for (const dependencyHandler of this.dependencyHandlerRegistry) {
+                try {
+                    const recordedPaths = DependencyEventBundler.getRecordedDependencyPaths(
+                        dependencyHandler.language,
+                        workspaceFolder.uri
+                    )
+                    if (!recordedPaths) {
+                        continue
+                    }
+                    dependencyHandler.updateDependencyMapBasedOnLSP(recordedPaths, workspaceFolder)
+                    ingestedDependencyCount += recordedPaths.length
+                } catch (error) {
+                    this.logging.debug(`Error ingesting dependency events for ${workspaceFolder.uri}: ${error}`)
+                }
+            }
+            this.dependencyEventsIngestedFolderUris.add(workspaceFolder.uri)
```

Comment:
this is marking workspaceFolder.uri as ingested even if some of the dependencies are failed or thrown errors.